src/libraries/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncEventArgs.Windows.cs

```diff
@@ -280,11 +262,10 @@ internal unsafe SocketError DoOperationAccept(Socket socket, SafeSocketHandle ha
                         out int bytesTransferred,
                         overlapped);
 
-                    return ProcessIOCPResultWithDeferredAsyncHandling(success, bytesTransferred, overlapped, buffer, cancellationToken);
+                    return ProcessIOCPResult(success, bytesTransferred, overlapped, buffer, cancellationToken);
                 }
                 catch
                 {
-                    _asyncProcessingState = AsyncProcessingState.None;
                     FreeNativeOverlapped(overlapped);
                     throw;
```

Comment:
I just wonder why some of the `catch`es have `_singleBufferHandle.Dispose();` inside and some don't (like this, even though there's a buffer passed to `ProcessIOCPResult` to potentially be pinned)

Reply:
None of them are needed anymore. Looks like I missed deleting one.